demo/api/src/importer/importers/product-importer/product-importer.ts

```diff
@@ -0,0 +1,54 @@
+import { EntityManager } from "@mikro-orm/core";
+import { Injectable, Logger, LoggerService } from "@nestjs/common";
+import { DataStream, DataStreamAndMetaData } from "@src/importer/data-streams/data-stream";
+import { pipeline, Transform } from "stream";
+
+@Injectable()
+export class ProductImporter {
+    dataStream: DataStreamAndMetaData | null = null;
+    name = "rawProductImport";
+    transformPipes: Transform[] = [];
+    logger: LoggerService;
+
+    constructor(private readonly em: EntityManager) {
+        this.logger = new Logger("product-importer");
+        this.transformPipes = [
+            new Transform({
+                objectMode: true,
+                transform: this.displayData.bind(this),
+            }),
+        ];
+    }
+
+    async init({ dataStream }: { dataStream: DataStream }): Promise<void> {
```

Comment:
so this version is a partially configured importer? because the source is still configurable...